golang/cli_test/cli_test.go

```diff
@@ -0,0 +1,448 @@
+    f := InitFixtures(t)
+	proc := f.RelayDStart()
+	defer proc.Stop(false)
+	fooAddr := f.KeyAddress(keyFoo)
+	genesisRelay := f.QueryGetRelayGenesis(fooAddr)
+	actual := genesisRelay.Res
+
+	// Condition
+	require.Equal(t, expected, actual)
+
+	//Cleanup
+	f.Cleanup()
+}
+
+func TestRelayCLIGetLastReorgLCA(t *testing.T) {
+	// Get Expected Value
+	var genesisHeaders []rtypes.BitcoinHeader
+	genesisJSON := readJSONFile(t, "genesis")
+	err := json.Unmarshal([]byte(genesisJSON), &genesisHeaders)
+	require.NoError(t, err)
+	expected := genesisHeaders[1].HashLE
+
+	// Query Chain for Actual Value
+	f := InitFixtures(t)
+	proc := f.RelayDStart()
+	defer proc.Stop(false)
+	fooAddr := f.KeyAddress(keyFoo)
+	lastReorgLCA := f.QueryGetLastReorgLCA(fooAddr)
+	actual := lastReorgLCA.Res
+
+	// Condition
+	require.Equal(t, expected, actual)
+
+	//Cleanup
+	f.Cleanup()
+}
+
+func TestRelayCLIGetBestDigest(t *testing.T) {
+	// Get Expected Value
+	var genesisHeaders []rtypes.BitcoinHeader
+	genesisJSON := readJSONFile(t, "genesis")
+	err := json.Unmarshal([]byte(genesisJSON), &genesisHeaders)
+	require.NoError(t, err)
+	expected := genesisHeaders[1].HashLE
+
+	// Query Chain for Actual Value
+	f := InitFixtures(t)
+	proc := f.RelayDStart()
+	defer proc.Stop(false)
+	fooAddr := f.KeyAddress(keyFoo)
+	bestDigest := f.QueryGetBestDigest(fooAddr)
+	actual := bestDigest.Res
+
+	// Condition
+	require.Equal(t, expected, actual)
+
+	//Cleanup
+	f.Cleanup()
+}
+
+func TestRelayCLIQueryFindAncestor(t *testing.T) {
+	// Extract data for transactions
+	var genesisHeaders []rtypes.BitcoinHeader
+	genesisJSON := readJSONFile(t, "genesis")
+	err := json.Unmarshal([]byte(genesisJSON), &genesisHeaders)
+	require.NoError(t, err)
+
+	var newDifficultyHeaders []rtypes.BitcoinHeader
+	newDiffJSON := readJSONFile(t, "0_new_difficulty")
+	err = json.Unmarshal([]byte(newDiffJSON), &newDifficultyHeaders)
+	require.NoError(t, err)
+
+	// Initialize chain
+	f := InitFixtures(t)
+	proc := f.RelayDStart()
+	defer proc.Stop(false)
+
+	// define paramater values
+	fooAddr        := f.KeyAddress(keyFoo)
+	prevEpochStart := hex.EncodeToString(genesisHeaders[0].HashLE[:])
+	digest         := hex.EncodeToString(newDifficultyHeaders[1].HashLE[:])
+	offset         := "5"
+
+	// ingest headers
+	success, stdout, stderr := f.TxIngestDiffChange(fooAddr, prevEpochStart, "0_new_difficulty.json", "--inputfile -y")
+	require.True(t, success, stderr)
+	require.Contains(t, stdout, `"success":true`)
+
+	// Require findancestor fails if ancestor does not exist on relay
+	f.QueryFindAncestorInvalid("could not find ancestor", digest, offset)
+
+	// Require findancestor returns ancestor if valid query
+	offset = "1"
+	findancestor := f.QueryFindAncestor(digest, offset)
+	expected := hex.EncodeToString(newDifficultyHeaders[0].HashLE[:])
+	actual := hex.EncodeToString(findancestor.Res[:])
+	require.Equal(t, expected, actual)
+}
+
+// func TestRelayCLIIsMostRecentCommonAncestor(t *testing.T) {
```

Comment:
derp gotta back to this

Reply:
is this going to be part of the finished PR?

Reply:
Will uncomment and add a test condition. Does that answer your question?

Reply:
ya :+1: